src/components/ModalComponent.tsx

```diff
@@ -139,13 +143,23 @@ export class ModalComponent extends React.Component<ModalComponentProps, any> {
                         }
                     </Alert>
                 }
-                <NestedEditComponent field={this.props.constants} modalType={this.props.modalType}
-                    readonly={this.props.constants.readonly === true}
-                    additionalModels={this.props.additionalModels} fetch={this.props.fetch}
-                    modelChanged={this.modelChanged} currentModel={this.state.item}
-                    showTitle={false}
-                    parentModel={{}}
-                />
+
+                {this.props.modalType === "BULK_CREATE" ?
+                        <FormGroup>
+                            <ControlLabel>Sync URL</ControlLabel>
+                            <FormControl type="text"
+                                         value={this.state.syncUrl}
+                                         placeholder="Sync CSV URL" onChange={this.syncUrl} />
+                        </FormGroup>
+                    : <NestedEditComponent
+                        field={this.props.constants} modalType={this.props.modalType}
+                        readonly={this.props.constants.readonly === true}
+                        additionalModels={this.props.additionalModels} fetch={this.props.fetch}
+                        modelChanged={this.modelChanged} currentModel={this.state.item}
+                        showTitle={false}
+                        parentModel={{}}
+                />}
```

Comment:
bulk creation is not using anything of the modal component, make a diff. modal & use that